ai.js

```diff
@@ -1,67 +1,65 @@
 const axios = require('axios');
 
 async function callOpenApi(prompt, model, token) {
-    let returnStr = "";
-
-    await axios.post('https://api.openai.com/v1/completions',
-    {
-        "model": model,
-        "prompt": prompt,
-        "max_tokens": 1024
-    }, 
-    {
-        headers: {
-            Authorization: "Bearer " + token
-        },
-        timeout: 10000
-    }).then(res => {
-        returnStr = res.data.choices[0].text.trim().substring(0, 2000);
+    try {
+        const res = await axios.post('https://api.openai.com/v1/completions', 
+            {
+                "model": model,
+                "prompt": prompt,
+                "max_tokens": 1024
+            }, 
+            {
+                headers: {
+                    Authorization: "Bearer " + token
+                },
+                timeout: 10000
+            }
+        );
+        const result = res.data.choices[0].text.trim().substring(0, 2000)
         console.log('==========')
         console.log(returnStr);
         console.log('==========')
-    }).catch(err => {
+        return result;
+    }
+    catch (err) {
         console.log("An error has occurred. Try again later")
         console.log(err)
-        returnStr = "Sorry, an internal error has occurred. Try again later.";
-    })
-
-    return returnStr;
+        return "Sorry, an internal error has occurred. Try again later.";
```

Comment:
I guess I could return the string values instead of using an intermediate variable. Is that generally "better form?"

Reply:
Returning immediately rather than relying on an intermediate value tends to reduce bugs because you miss a break or an `else` before another `if` statement. It's obvious where the function exits.

Reply:
The other benefit you get here is immutability -- less mutation means fewer bugs, generally.